core/src/action.rs

```diff
@@ -72,23 +72,74 @@ impl Hash for ActionWrapper {
 /// All Actions for the Holochain Instance Store, according to Redux pattern.
 #[derive(Clone, PartialEq, Debug)]
 pub enum Action {
-    /// entry to Commit
-    /// MUST already have passed all callback checks
+    // ----------------
+    // Agent actions:
+    // ----------------
+    /// Writes an entry to the source chain.
+    /// Does not validate, assumes entry is valid.
     Commit(Entry),
+
+    // -------------
+    // DHT actions:
+    // -------------
+    /// Adds an entry to the local DHT shard.
+    /// Does not validate, assumes entry is valid.
+    Hold(Entry),
+
+    /// Adds a link to the local DHT shard's meta/EAV storage
+    /// Does not validate, assumes link is valid.
+    AddLink(Link),
+
+    // ----------------
+    // Network actions:
+    // ----------------
+    /// Create a network proxy instance from the given JSON config.
+    /// 2nd and 3rd parameter are the DNA hash and the agent id
+    /// which are needed to register with the network.
+    InitNetwork((JsonString, String, String)),
+
+    /// Makes the network PUT the given entry to the DHT.
+    /// Distinguishes between different entry types and does
+    /// the right thing respectively.
+    /// (only publish for AppEntryType, publish and publish_meta for links etc)
+    Publish(Address),
+
     /// GetEntry by address
     GetEntry(Address),
     GetEntryTimeout(Address),
 
-    /// link to add
-    AddLink(Link),
-    /// get links from entry address and attribute-name
-    //GetLinks(GetLinksArgs),
+    /// Lets the network module respond to a GET request.
+    /// Triggered from the corresponding workflow after retrieving the
+    /// requested entry from our local DHT shard.
+    RespondGet((GetDhtData, Option<Entry>)),
 
-    /// execute a function in a zome WASM
-    ExecuteZomeFunction(ZomeFnCall),
-    /// return the result of a zome WASM function call
-    ReturnZomeFunctionResult(ExecuteZomeFnResponse),
+    /// We got a response for our GET request which needs to be
+    /// added to the state.
+    /// Triggered from the network handler.
+    HandleGetResult(DhtData),
 
+    /// Sends a direct message object to the given address.
+    /// 3rd parameter is the message id
+    /// 4th parameter is true for a response to a previous message, false for a new interaction
+    SendDirectMessage((Address, DirectMessage, String, bool)),
```

Comment:
It may be time ditch the tuple in favor of a sub-struct so the parameters can be named.

```rust
pub struct DirecmMessageData {
    address: Address,
    message: DirectMessage,
    msg_id: String,
    continue: bool,
}

pub enum Action {
    SendDirectMessage(DirectMessageData)
```